components/SplitPane/library/src/desktopMain/kotlin/org/jetbrains/compose/splitpane/DesktopSplitPane.kt

```diff
@@ -51,101 +53,84 @@ internal actual fun SplitPane(
             val secondMinSizePx = secondPlaceableMinimalSize.value * density
 
             with(splitPaneState) {
-                val constrainedMin = constraints.minByDirection(isHorizontal) + firstMinSizePx
-                val constrainedMax =
-                    (constraints.maxByDirection(isHorizontal).toFloat() - secondMinSizePx).let {
-                        if (it <= 0 || it <= constrainedMin) {
-                            constraints.maxByDirection(isHorizontal).toFloat()
-                        } else {
-                            it
-                        }
-                    }
+                val firstMeasurable = measurables[0]
+                val splitterMeasurable = measurables[1]
+                val secondMeasurable = measurables[2]
+                val handleMeasurable = measurables[3]
 
-                if (minPosition != constrainedMin) {
-                    maxPosition = constrainedMin
-                }
+                // Need the size of the splitter to determine the min/max position
+                // Constrain the splitter only on the "other" axis
+                val splitterConstraints =
+                    if (isHorizontal)
+                        constraints.withUnconstrainedWidth()
+                    else
+                        constraints.withUnconstrainedHeight()
+                val splitterPlaceable = splitterMeasurable.measure(splitterConstraints)
+                val splitterSize = splitterPlaceable.valueByDirection(isHorizontal)
 
-                if (maxPosition != constrainedMax) {
-                    maxPosition =
-                        if (firstMinSizePx + secondMinSizePx < constraints.maxByDirection(isHorizontal)) {
-                            constrainedMax
-                        } else {
-                            minPosition
-                        }
-                }
+                @Suppress("UnnecessaryVariable")
+                val constrainedMin = firstMinSizePx
+                val maxConstraintOnMainAxis = constraints.maxByDirection(isHorizontal)
+                val constrainedMax = (maxConstraintOnMainAxis - secondMinSizePx - splitterSize)
+                    .coerceAtLeast(constrainedMin)
 
-                val constrainedPosition =
-                    (constraints.maxByDirection(isHorizontal) - (firstMinSizePx + secondMinSizePx)).let {
-                        if (it > 0f) {
-                            (it * positionPercentage).coerceIn(constrainedMin, constrainedMax).roundToInt()
-                        } else {
-                            constrainedMin.roundToInt()
-                        }
-                    }
+                minPosition = constrainedMin
+                maxPosition = constrainedMax
 
+                val position = (constrainedMin * (1-positionPercentage) + constrainedMax * positionPercentage)
```

Comment:
For me personally `position = (min + percentage * (max - min)).roundToInt()` is more clear, so we handle percentage only on the handleArea